### DISCUSSION TITLE
Truncating the locals in the error stack trace

### DISCUSSION THREAD

**Discussion by RishiMalhotra920** [fastapi/typer#923] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import torch
import typer

app = typer.Typer()


@app.command()
def main(size: int = typer.Option(100, help="Size of the tensor")):
    """
    Create a large tensor and raise an exception to demonstrate output flooding.
    """
    # Create a large tensor
    large_tensor = torch.randn(size, size, size, size)
    large_tensor_2 = torch.randn(size, size, size, size)
    large_tensor_3 = torch.randn(size, size, size, size)

    # Function that will raise an exception
    def cause_exception(tensor):
        print("About to raise an exception. Local variables will include the large tensor.")
        raise ValueError("This exception will cause the large tensor to be printed")

    # Call the function that raises an exception
    cause_exception(large_tensor)


if __name__ == "__main__":
    app()
```

### Description

you will need to run `pip install torch`

Try running the example code above using python app.py. You will notice that the large tensors flood the terminal stack trace. Is there a way to shorten the length of these local variable values in the stack trace. I don't want to turn these off completely but these long tensors in my code make the stack trace unreadable!

Alternatively, you can allow us to use custom rich Console variables with typer. I did some digging and the code below should do the trick. All we need is a way for us to use these custom Consoles with typer.

```
console = Console(width=80)
install(console=console, show_locals=True, max_frames=10, locals_max_length=5, locals_max_string=100)
```

Perhaps, we can add this in the `__init__.py` for typer.Typer() eg:

```
app = typer.Typer(rich_console=console)
```

I'd be satisfied with either of these options. The first approach (of including an option to truncate the length) seems easier to implement. Thanks a lot!!

### Operating System

macOS

### Typer Version

0.12.3

### Python Version

3.10.14

### Additional Context

![image](https://github.com/user-attachments/assets/df97cfa1-cffb-4c37-a076-3c1a347e0e39)

Some images of the terminal ^

**Comment by RishiMalhotra920** [comment]
Nevermind, i found that the code i wrote using the rich library also does not truncate the pytorch variables in the right way so allowing custom consoles would not help. I'm curious if anyone else here has any solutions!

**Reply by Habush** [reply, marked as answer]
Facing similar issue. The current error output makes it difficult to navigate to the source of an error. Did you find a solution?

EDIT: Passing `pretty_exceptions_enable=False` when initializing `typer` object fixes it for me. [Source](https://stackoverflow.com/a/76375308)

**Reply by avicennax** [reply]
^ this marked answer doesn't solve issue under the assumption you still want prettified output. Relatedly, I think it might be useful to allow users to specify which locals they want to obfuscated to avoid blowing up the output.

Related to: https://typer.tiangolo.com/tutorial/exceptions/#disable-local-variables-for-security, but allow users to glob match on locals to filter (or either name or class).
